athenahealth/lab_results.go

```
@@ -0,0 +1,303 @@
+
+const (
+	LabResultAttachmentTypeBMP  LabResultAttachmentType = "BMP"
+	LabResultAttachmentTypeGIF  LabResultAttachmentType = "GIF"
+	LabResultAttachmentTypeJPG  LabResultAttachmentType = "JPG"
+	LabResultAttachmentTypeJPEG LabResultAttachmentType = "JPEG"
+	LabResultAttachmentTypePDF  LabResultAttachmentType = "PDF"
+	LabResultAttachmentTypeTIF  LabResultAttachmentType = "TIF"
+	LabResultAttachmentTypeTIFF LabResultAttachmentType = "TIFF"
+)
+
+type AddLabResultDocumentOptions struct {
+	// AttachmentContents must be Base64 encoded
+	AttachmentContents io.Reader               `json:"attachmentcontents"`
+	AttachmentType     LabResultAttachmentType `json:"attachmenttype"`
+	InternalNote       *string                 `json:"internalnote"`
+	NoteToPatient      *string                 `json:"notetopatient"`
+	ObservationDate    *time.Time              `json:"observationdate"`
+	OriginalFilename   *string                 `json:"originalfilename"`
+	// 1 = high, 2 = normal
+	Priority    *string `json:"priority"`
+	ResultNotes *string `json:"resultnotes"`
+	// Final, Partial, Pending, Preliminary, Corrected, Cancelled
+	ResultStatus *string `json:"resultstatus"`
+	TieToOrderID *int    `json:"tietoorderid"`
+}
+
+type addLabResultDocumentResponse struct {
+	ErrorMessage string `json:"errormessage"`
+	LabResultID  int    `json:"labresultid"`
+	Success      bool   `json:"success"`
+}
+
+// AddLabResultDocument creates a lab result document record of a specific patient
+//
+// POST /v1/{practiceid}/patients/{patientid}/documents/labresult
+//
+// https://docs.athenahealth.com/api/api-ref/document-type-lab-result#Add-lab-result-document-to-patient's-chart
+func (h *HTTPClient) AddLabResultDocumentReader(ctx context.Context, patientID string, departmentID string, opts *AddLabResultDocumentOptions) (int, error) {
+	form := NewFormURLEncoder()
+
+	form.AddString("departmentid", departmentID)
+
+	if opts != nil {
+		if opts.AttachmentContents != nil {
```

we may want to surface an error if the attachment contents reader is `nil`

---

This gets into some divergence between how the athena API / client thinks about lab results and Eleanor does -- for us yes that should be an error case. For athena though it's seems valid to upload lab results with no attachments, for example if analytes and a result/interpretation are provided. Another one that might be good to chat more about.

---

Discussed and reviewed API and confirmed these are not required fields for this API so will not include error handling for those